src/auxiliary/t8code.jl

```diff
@@ -0,0 +1,452 @@
+        neighbor_ielements = unsafe_wrap(Array,pelement_indices_ref[],num_neighbors)
+        neighbor_leafs     = unsafe_wrap(Array,pneighbor_leafs_ref[],num_neighbors)
+        neighbor_scheme    = pneigh_scheme_ref[]
+
+        if num_neighbors > 0
+          neighbor_level = t8_element_level(neighbor_scheme, neighbor_leafs[1])
+
+          # Conforming interface: The second condition ensures we only visit the interface once.
+          if level == neighbor_level && current_index <= neighbor_ielements[1]
+          # TODO: Find a fix for the case: Single element on root level with periodic boundaries.
+          # elseif level == neighbor_level &&
+          #   (all(Int32(current_index) .< neighbor_ielements) ||
+          #   level == 0 && (iface == 0 || iface == 2 || iface == 4))
+              local_num_conform += 1
+
+              faces = (iface, dual_faces[1])
+              interface_id = local_num_conform
+
+              # Write data to interfaces container.
+              interfaces.neighbor_ids[1, interface_id] = current_index + 1
+              interfaces.neighbor_ids[2, interface_id] = neighbor_ielements[1] + 1
+
+              # Iterate over primary and secondary element.
+              for side = 1:2
+                # Align interface in positive coordinate direction of primary element.
+                # For orientation == 1, the secondary element needs to be indexed backwards
+                # relative to the interface.
+                if side == 1 || orientation == 0
+                  # Forward indexing
+                  indexing = :i_forward
+                else
+                  # Backward indexing
+                  indexing = :i_backward
+                end
+
+                if faces[side] == 0
+                  # Index face in negative x-direction
+                  interfaces.node_indices[side, interface_id] = (:begin, indexing)
+                elseif faces[side] == 1
+                  # Index face in positive x-direction
+                  interfaces.node_indices[side, interface_id] = (:end, indexing)
+                elseif faces[side] == 2
+                  # Index face in negative y-direction
+                  interfaces.node_indices[side, interface_id] = (indexing, :begin)
+                else # faces[side] == 3
+                  # Index face in positive y-direction
+                  interfaces.node_indices[side, interface_id] = (indexing, :end)
+                end
+              end
+
+          # Non-conforming interface.
+          elseif level < neighbor_level 
+              local_num_mortars += 1
+
+              faces = (dual_faces[1],iface)
+
+              mortar_id = local_num_mortars
+
+              # Last entry is the large element.
+              mortars.neighbor_ids[end, mortar_id] = current_index + 1
+
+              # First `1:end-1` entries are the smaller elements.
+              mortars.neighbor_ids[1:end-1, mortar_id] .= neighbor_ielements[:] .+ 1
```

Comment:
```suggestion
              mortars.neighbor_ids[1:end-1, mortar_id] .= neighbor_ielements .+ 1
```
This looks like an unnecessary copy on the right-hand side.